src/main/kotlin/org/rust/cargo/runconfig/test/CargoTestEventsConverter.kt

```diff
@@ -257,10 +258,22 @@ class CargoTestEventsConverter(
             ServiceMessageBuilder.testIgnored(test.name)
                 .addAttribute("nodeId", test)
 
+        private fun createTestStdOutMessage(test: NodeId, stdout: String): ServiceMessageBuilder =
+            ServiceMessageBuilder.testStdOut(test.name)
+                .addAttribute("nodeId", test)
+                .addAttribute("out", stdout)
+
         private fun createTestCountMessage(testCount: String): ServiceMessageBuilder =
             ServiceMessageBuilder("testCount")
                 .addAttribute("count", testCount)
 
+        private fun parseFailedTestOutput(output: String): FailedTestOutput {
+            val partitionPredicate: (String) -> Boolean = { !it.trimStart().startsWith("thread", true) }
+            val stdout = output.lineSequence().takeWhile(partitionPredicate).joinToString("\n")
+            val failedMessage = output.lineSequence().dropWhile(partitionPredicate).joinToString("\n")
```

Comment:
Looks like you can use `Sequence.partition` here

Reply:
This name is deceptive :)
> Splits the original sequence into pair of lists, where first list contains elements for which predicate yielded `true`, while second list contains elements for which predicate yielded `false`.